### PR TITLE
[WIP] Support surface-based analysis in FSL

### PR DESCRIPTION
**Pull request by effigies**
Not sure how huge the scope of this might get. Hopefully it's just some minor tweaks, because my FSL knowledge is super brittle.

### CODE REVIEW

**Comment by satra** [comment]
@effigies - so film_gls needs a gifti with timeseries data right? do you know if it supports cifti directly?

```markdown
	**--in <file>**	input data file ( NIFTI for volumetric, **GIFTI for surface** )

Optional arguments (You may optionally specify one or more of):
	--ac		perform autocorrelation estimation only
	--thr <num>	initial threshold to apply to input data
	--ar		fits autoregressive model - default is to use tukey with M=sqrt(numvols)
	--help		prints this message
	--noest		do not estimate auto corrs
	--outputPWdata		output prewhitened data and average design matrix
	--pava		estimates autocorr using PAVA - default is to use tukey with M=sqrt(numvols)
	--sa		smooths auto corr estimates
	-v		outputs full data
	--rn <file>	directory name to store results in, default is results
	--mode <mode>	analysis mode, options are volumetric ( default ) or surface. Caution: surface-based functionality is still BETA
	**--in2 <file>**	input surface for autocorr smoothing in surface-based analyses
	--mf <file>	re-estimate mean_func baseline - for use with perfusion subtraction
	--mft <file>	minimum timepoint file
	--pd <file>	paradigm file
	--con <file>	t-contrasts file
	--fcon <file>	f-contrasts file
	***--epith <num>**	susan brightness threshold for volumetric analysis/smoothing sigma for surface analysis
	**--ms <num>**	susan mask size for volumetric analysis/smoothing extent for surface analysis
	--tukey <num>	uses tukey window to estimate autocorr with window size num - default is to use tukey with M=sqrt(numvols)
	--mt <num>	uses multitapering with slepian tapers and num is the time-bandwidth product - default is to use tukey with M=sqrt(numvols)
	--ven		list of numbers indicating voxelwise EVs position in the design matrix (list order corresponds to files in vxf option). Caution BETA option, only use with volumetric analysis.
	--vef		list of 4D images containing voxelwise EVs (list order corresponds to numbers in vxl option). Caution BETA option, only use with volumetric analysis.
```

it may be good to ping the FSL folks for some examples here.

**Comment by effigies** [comment]
I haven't tried it. I've actually had difficulty getting it to run at all with GIFTI, so this is a little stalled until I have a bit more time to focus on it.

File: nipype/interfaces/fsl/model.py [hunk 1]
```diff
@@ -349,7 +349,11 @@ def _run_interface(self, runtime):
                                                       self.inputs.contrasts,
                                                       do_tempfilter, basis_key)
             nim = load(func_files[i])
-            (_, _, _, timepoints) = nim.shape
+            if hasattr(nim, 'get_arrays_from_intent'):
```
Comment (oesteban) [inline nipype/interfaces/fsl/model.py]: What about: `timepoints = getattr(nim, 'get_arrys_from_intent', nim.shape[3])`?

**Comment by effigies** [comment]
This pull request is "orphaned," which means it has been deemed to be abandoned by its original author. Orphaned pull requests have not been rejected, and we hope that if a user sees one that will meet their needs with a little work, that they will fork it and open a new pull request (or, in the case of the original author, reopen the original PR).

We ask that all adopted PRs be updated to merge or rebase the current master. If you would like to adopt a PR and need help getting started, any of a number of contributors will be happy to help.

**Pull request closed without merging**